hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTable.java

```diff
@@ -766,6 +767,10 @@ protected void reconcileAgainstMarkers(HoodieEngineContext context,
           waitForAllFiles(context, invalidPathsByPartition, FileVisibility.DISAPPEAR);
         }
       }
+      if (config.isFailRetriesAfterFinalizeWrite()) {
```

Comment:
In general, there is no good reason for us to finalize the write of the data table, then write to MT table, I feel? Would n't that cut down the number of times you hit these issues, with MT `files` partition filtering out files written by any stray executors?